packages/nimble-components/src/mapping/base/index.ts

```diff
@@ -0,0 +1,37 @@
+import { attr, nullableNumberConverter } from '@microsoft/fast-element';
+import { FoundationElement } from '@microsoft/fast-foundation';
+import type { MappingConfig } from '../../table-column/enum-base';
+import type { MappingKeyType } from '../../table-column/enum-base/types';
+
+/**
+ * An element to be given as content to a nimble-table-column-mapping or nimble-table-column-icon.
+ * This is a base class from which other specific mapping types (e.g. MappingIcon, MappingText) are derived.
+ */
+export abstract class Mapping extends FoundationElement {
+    @attr({ mode: 'fromView' })
+    public key?: string | number | boolean;
+
+    @attr({ attribute: 'default-mapping', mode: 'boolean' })
+    public defaultMapping = false;
+
+    public static typeConvertKey(
```

Comment:
Think it would be good to document this public method.

Reply:
Added documentation. I also think that it makes sense to leave it in the `Mapping` class, since it is primarily for the mappings to use.